source/common/json/config_schemas.cc

```diff
@@ -448,6 +448,7 @@ const std::string Json::Schema::ROUTE_ENTRY_CONFIGURATION_SCHEMA(R"EOF(
       "prefix" : {"type" : "string"},
       "path" : {"type" : "string"},
       "cluster" : {"type" : "string"},
+      "cluster_header" : {"type" : "string"},
       "weighted_clusters": {"$ref" : "#/definitions/weighted_clusters"},
       "host_redirect" : {"type" : "string"},
```

Comment:
I already discussed this with @ccaraman. AFAICT, there is no way to use oneof in this context in a JSON schema (intermixed with properties inside an object). If you know otherwise let me know. Same comment applies to below.